edgelet/edgelet-core/src/module.rs

```diff
@@ -366,12 +366,33 @@ pub struct SystemInfo {
     pub architecture: String,
     /// iotedge version string
     pub version: &'static str,
+    pub provisioning: ProvisioningInfo,
     pub server_version: String,
     pub kernel_version: String,
     pub operating_system: String,
     pub cpus: i32,
 }
 
+#[derive(Clone, Debug, Default, Serialize)]
+pub struct ProvisioningInfo {
+    /// IoT Edge provisioning type, examples: manual.device_connection_string, dps.x509
+    pub r#type: String,
```

Comment:
why is this r#type instead of type?

Reply:
Because `type` is a reserved word.

Reply:
(The alternative is `#[serde(rename = "type")] pub type_: String`, but since we have edition 2018 we might as well take advantage of it.)